text/java/0000-low-profile-memory-calc.md

```diff
@@ -0,0 +1,103 @@
+
+The minimum value for heap is 2M (Java requirement). This means that you can get into some situations where the memory calculator will do it's best to configure the JVM, but the only configuration it can create does not provide sufficient heap space for your application. At this point, a user would need to take control and manually reduce the thread stack size, thread count or code cache size such that there is sufficient memory remaining. It is strongly recommend that you do not set the heap size directly, i.e. `-Xmx` because the memory calculator will do this automatically to whatever free memory is available after all other JVM memory regions are taken into account. In addition if you set `-Xmx` too high you can make memory calculator fail and too low will leave unused memory in the container.
+
+The minimum selection with this algorithm would be: `256k * 30 threads + 2.5M code cache + 10M direct = 20M` plus metaspace and heap. This compares to `500M` plus metaspace and heap for the current memory calculator implementation.
+
+Here are some example calculations based on the rules above:
+
+| thread stack | thread count | code cache | metaspace [1] | heap [2] | direct | container memory | Notes                                                                                |
+| ------------ | ------------ | ---------- | ------------- | -------- | ------ | ---------------- | ------------------------------------------------------------------------------------ |
+| 1M           | 250          | 240M       | 100M          | 424M     | 10M    | 1G               | Baseline                                                                             |
+| 512K         | 125          | 120M       | 100M          | 220M     | 10M    | 512M             | Container memory reduced by 1/2 the baseline                                         |
+| 256K         | 62           | 60M        | 100M          | 70M      | 10M    | 256M             | Container memory reduced by 1/4 the baseline                                         |
+| 384K         | 250          | 240M       | 100M          | 580M     | 10M    | 1G               | Baseline, user has fixed thread stack at 384K                                        |
+| 384K         | 125          | 120M       | 100M          | 235M     | 10M    | 512M             | Container memory reduced by 1/2 the baseline, user has fixed thread stack at 384K    |
+| 384K         | 62           | 60M        | 100M          | 62M      | 10M    | 256M             | Container memory reduced by 1/4 the baseline, user has fixed thread stack at 384K    |
+| 1M           | 50           | 240M       | 100M          | 624M     | 10M    | 1G               | Baseline, user has fixed thread count to 50                                          |
+| 512K         | 50           | 120M       | 100M          | 257M     | 10M    | 1G               | Container memory reduced by 1/2 the baseline, user has fixed thread count to 50      |
```

Comment:
I think the Container Memory column from here hasn't been updated to reflect the notes?

Reply:
Good catch. Updated.